spark/src/main/java/org/apache/iceberg/spark/SparkDataFile.java

```diff
@@ -46,18 +47,38 @@
   private final int upperBoundsPosition;
   private final int keyMetadataPosition;
   private final int splitOffsetsPosition;
+  private final int specIdPosition;
   private final Type lowerBoundsType;
   private final Type upperBoundsType;
   private final Type keyMetadataType;
 
   private final SparkStructLike wrappedPartition;
+  private final Types.StructType partitionStruct;
   private Row wrapped;
 
+  private static final StructLike EMPTY_PARTITION_INFO = new StructLike() {
+    @Override
+    public int size() {
+      return 0;
+    }
+
+    @Override
+    public <T> T get(int pos, Class<T> javaClass) {
+      return null;
```

Comment:
If size is 0, then any call to `get` or `set` is invalid. In that case, I think it is safer to throw an exception here.

Reply:
Sgtm, mirrored this in the "set" command, might as well throw in both.